File: pascal.py
def pascal(k):
	prev, cur = [], []
	
	if(k == 0):
		return [1]
	if(k == 1):
		return [1, 1]

	prev = [1, 1]

	for j in range(2, k+1):
		cur = [1]

		for i in range(1, j):
			res = prev[i-1] + prev[i]
			cur.append(res)

		cur.append(1)

		prev = cur


	return cur
	
res = pascal(4)
print(res)

File: pascal.cpp
#include <bits/stdc++.h>

using namespace std;


vector<int> printPascal(const int k){
	if(k == 0)
		return vector<int> {1};
	else if(k==1)
		return vector<int> {1, 1};
	else{
		vector<int> prev{1, 1};

		for(int i=2; i<=k; ++i){
			vector<int> res{1};

			for(int j=1; j<=i-1; ++j){
				res.push_back(prev[j-1] + prev[j]);
			}

			res.push_back(1);

			prev = res;
		}
		
		return prev;
	}
}



int main(){
	const int k = 5;

	auto res = printPascal(k);

	for(auto e : res)
		printf("%d ", e);
	cout << endl;

	return 0;
}